zerver/lib/events.py

```diff
@@ -1972,3 +1974,8 @@ def post_process_state(
             handle_stream_notifications_compatibility(
                 user_profile, stream_dict, notification_settings_null
             )
+
+    if not allow_empty_topic_name and "user_topics" in ret:
+        for user_topic in ret["user_topics"]:
+            if user_topic["topic_name"] == "":
+                user_topic["topic_name"] = Message.EMPTY_TOPIC_FALLBACK_NAME
```

Comment:
I don't understand this block -- can you explain why we need a block here? Do the tests fail if this code is deleted?

I'd have expected that this would be handled just via the earlier code; this post-processing function is meant to be only for things like translating a `dict` to ` list` for the API... and I'd have thought the reformatting would be done earlier.

Since we know the event queue will have the same client capability value as the `fetch_initial_state_data` caller, I think just changing the initial population code should be sufficient, and `apply_events` will do the right thing via only processing topics in the same namespace,

Reply:
That'll also work but it'll lead to more compatibility code additions within other lib functions.

For example, if we add this logic in `get_user_topics` (the initial population code) then it'll affect `get_topic_mutes` which is used in `bulk_do_set_user_topic_visibility_policy` -- so, we'll have to add some compatibility code in `bulk_do_set_user_topic_visibility_policy` which is used at a lot of places and will lead to a lot of changes.

So, I think just adding it in the last stage of the flow is the right approach ?